### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by andreasbeer1981** [contentful/contentful.rb#49]
Shouldn't client.rb raise an exception on getting a timeout response?

```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html>
    <head>
        <title>503 Connection timed out</title>
    </head>
    <body>
        <h1>Error 503 Connection timed out</h1>
        <p>Connection timed out</p>
        <h3>Guru Mediation:</h3>
        <p>Details: ...</p>
        <hr>
            <p>Varnish cache server</p>
    </body>
</html>
```

**Comment by pxlpnk** [comment]
this depends on how you initialised the client. To raise exceptions you need to set `raise_errors: true` 
See the [documentation](https://github.com/contentful/contentful.rb#raise_errors).

The interesting part is though, that you should not run into this kind of error at all. 
Would you mind sending an email to support@contentful.com with the information on what calls you received the error and around what time?

**Comment by andreasbeer1981** [comment]
mail is sent. thanks for the feedback

**Comment by pxlpnk** [comment]
Thank you,
I will add a better error handling for the 503 as this is returned as HTML instead of JSON.

**Comment by andreasbeer1981** [comment]
Hi,

we experienced the very same error again on:
February 17th 2015, 14:04:16
February 17th 2015, 14:14:29
February 17th 2015, 17:23:26

maybe https://github.com/contentful/contentful.rb/pull/50 still left some cases out?

thanks

**Comment by andreasbeer1981** [comment]
oh, #50 is not released yet, never mind :D

**Comment by pxlpnk** [comment]
@andreasbeer1981 #50 is already released with v0.5.0 
see the diff to master: https://github.com/contentful/contentful.rb/compare/v0.5.0...master

Could you provide more details and explain 
> left some cases out?
